src/lib/network_pool/transaction_pool.ml

```diff
@@ -968,385 +935,114 @@ struct
+        match e with
+        | Insufficient_replace_fee (`Replace_fee rfee, fee) ->
+            log
+              "rejecting $cmd because of insufficient replace fee ($rfee > \
+               $fee)"
+              ~metadata:
+                [ ("cmd", User_command.to_yojson tx)
+                ; ("rfee", Currency.Fee.to_yojson rfee)
+                ; ("fee", Currency.Fee.to_yojson fee)
+                ]
+        | Unwanted_fee_token fee_token ->
+            log "rejecting $cmd because we don't accept fees in $token"
+              ~metadata:
+                [ ("cmd", User_command.to_yojson tx)
+                ; ("token", Token_id.to_yojson fee_token)
+                ]
+        | _ ->
+            ()
 
-      let verify' ~allow_failures_for_tests (t : pool)
-          (diffs : t Envelope.Incoming.t) :
+      let verify (t : pool) (diff : t Envelope.Incoming.t) :
           verified Envelope.Incoming.t Deferred.Or_error.t =
-        let open Deferred.Let_syntax in
-        let trust_record =
-          Trust_system.record_envelope_sender t.config.trust_system t.logger
-            diffs.sender
+        let open Deferred.Or_error.Let_syntax in
+        let ok_if_true cond ~error =
+          Deferred.return
+            (Result.ok_if_true cond ~error:(Error.of_string error))
         in
-        let config = Indexed_pool.config t.pool in
-        let global_slot = Indexed_pool.global_slot_since_genesis t.pool in
-        let pool_max_size = t.config.pool_max_size in
-        let sender = Envelope.Incoming.sender diffs in
-        let is_sender_local = Envelope.Sender.(equal sender Local) in
-        let diffs_are_valid () =
-          List.for_all (Envelope.Incoming.data diffs) ~f:(fun cmd ->
-              let is_valid = not (User_command.has_insufficient_fee cmd) in
-              if not is_valid then
-                [%log' debug t.logger]
-                  "Filtering user command with insufficient fee from \
-                   transaction-pool diff $cmd from $sender"
-                  ~metadata:
-                    [ ("cmd", User_command.to_yojson cmd)
-                    ; ( "sender"
-                      , Envelope.(Sender.to_yojson (Incoming.sender diffs)) )
-                    ] ;
-              is_valid )
+        let%bind () =
+          (* TODO: we should probably remove this -- the libp2p gossip cache should cover this already *)
+          let (`Already_mem already_mem) =
+            Lru_cache.add t.recently_seen (Lru_cache.T.hash diff.data)
+          in
+          ok_if_true (not already_mem) ~error:"Recently seen"
         in
-        let h = Lru_cache.T.hash diffs.data in
-        let (`Already_mem already_mem) = Lru_cache.add t.recently_seen h in
-        if (not allow_failures_for_tests) && already_mem && not is_sender_local
-        then
-          (* We only reject here if the command was from the network: the user
-             may want to re-issue a transaction if it is no longer being
```

Comment:
I updated the predicate in my new code to be the same.

Reply:
I actually have a TODO in the code to just remove this LRU completely as I'm pretty certain we never hit this code (libp2p already has a cache for gossip messages, and its more powerful than this one). I'm just leaving this intact for now just in case it does get hit under rare scenarios.